File: Python/TestFor8.py
def test_for_8(num):

    flag = 0
    for j in range(3):
        num,mod = divmod(num,2)
        flag = flag + mod

    if flag == 0:
        return True
    else:
        return False


# Test to check that the function behaves properly
for i in range(100):
    print("{} divisible by 8 {}".format(i, test_for_8(i)))

File: Java/TestFor8.java
import java.io.*;
import java.util.*;

class TestFor8
{
    public static void main(String args[])
    {
        Scanner x =new Scanner(System.in);
        int num=0;
        System.out.print("Enter a number..:");
        int n = x.nextInt();
        int n2 = n;
        int temp =(int) n2 / 8;
        for(int i=1;i<=temp;i++)
        {
            num=num+8;
        }
        if(num == n2)
        {
            System.out.println(n2 +" is divisible by 8!!");
        }
        else
        {
            System.out.println(n2 +" is Not divisible by 8!!");
        }
    }
}